### PR TITLE
fix: Allow for usage of special characters in keyframes name

### PR DESCRIPTION
**Pull request by william-will-angi**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -321,7 +321,7 @@ function localizeDeclaration(declaration, context) {
   const isAnimation = /animation$/i.test(declaration.prop);
 
   if (isAnimation) {
-    const validIdent = /^-?[_a-z][_a-z0-9-]*$/i;
+    const validIdent = /^-?(?:\\[~!@$%^&*()+=,./';:"?><[]\\{}|`#]|[_a-z])(?:\\[~!@$%^&*()+=,./';:"?><[]\\{}|`#]|[_a-z0-9-])*$/i;
```
Comment (alexander-akait) [inline src/index.js:324]: Can you show where you get this regexp :smile: Ident token from spec https://www.w3.org/TR/css-syntax-3/#typedef-ident-token

Comment (william-will-angi) [reply src/index.js:324]: That spec seems to only explain escaping Unicode points (ie `\00002b`)

They have a diagram labeled `escape` which seems to indicate the character following the escape can be any character except newline - the spec explicitly says that the diagrams are not 100% accurate though, so I'm not sure how much stock I should put in that.

I got this list from the following post:
https://mathiasbynens.be/notes/css-escapes

Cross referencing some SO posts seemed to confirm this list. 

Would it be better to have a more relaxed regex around escapes?

Comment (alexander-akait) [reply src/index.js:324]: I am fine with it, just want to verify

Comment (william-will-angi) [reply src/index.js:324]: Is there anything else you would like me to address before getting these changes in?

Comment (william-will-angi) [reply src/index.js:324]: @alexander-akait I hope I'm not bothering. Is this MR ready to go?

Comment (william-will-angi) [reply src/index.js:324]: Would limiting the scope of these changes to the special character `@` speed this MR along? That is the only character currently affecting  our projects.

Comment (alexander-akait) [reply src/index.js:324]: Sorry for delay, yes, I think let's do it to avoid big perf regression

**Pull request closed without merging**